zingolib/src/lightclient.rs

```diff
@@ -946,16 +950,89 @@ impl LightClient {
             "outgoing_metadata" => outgoing_json,
         }
     }
-    /*
-        pub async fn do_list_txsummaries(&self) -> JsonValue {
-            for (txid, transaction_md) in self.wallet.transaction_context.transaction_metadata_set.read().await.current.iter()
-            {
-                let tx_value = transaction_md.total_value_spent();
-                let to_addresses =
-    }
-            Vec<TxSummary>
+    pub async fn do_list_txsummaries(&self) -> Vec<ValueTransferSummary> {
+        let mut sends = Vec::new();
+        let mut receipts = Vec::new();
+
+        for (txid, transaction_md) in self
+            .wallet
+            .transaction_context
+            .transaction_metadata_set
+            .read()
+            .await
+            .current
+            .iter()
+        {
+            let tx_value_spent = transaction_md.total_value_spent();
+            let tx_value_received = transaction_md.total_value_received();
+            let tx_change_received = transaction_md.total_change_returned();
+            let TransactionMetadata {
+                block_height,
+                datetime,
+                zec_price,
+                ..
+            } = transaction_md;
+            match (tx_value_spent, (tx_value_received - tx_change_received)) {
```

Comment:
This `match` statement is the "thinking" part of this machine.   Can we promote it to have it's own name.   Make it a fn?